base_layer/core/src/transactions/transaction_components/transaction_input.rs

```diff
@@ -168,15 +168,31 @@ impl TransactionInput {
         script_public_key: &PublicKey,
         commitment: &Commitment,
     ) -> [u8; 32] {
+        let message = TransactionInput::build_script_signature_message(version, script, input_data, commitment);
         match version {
             TransactionInputVersion::V0 | TransactionInputVersion::V1 => {
                 DomainSeparatedConsensusHasher::<TransactionHashDomain>::new("script_challenge")
                     .chain(&version)
                     .chain(ephemeral_commitment)
                     .chain(ephemeral_pubkey)
+                    .chain(script_public_key)
+                    .chain(&message)
+                    .finalize()
+            },
+        }
+    }
+
+    pub fn build_script_signature_message(
+        version: TransactionInputVersion,
+        script: &TariScript,
+        input_data: &ExecutionStack,
+        commitment: &Commitment,
```

Comment:
It makes sense why `script` and `input_data` are hashed here for complexity reasons, but why `commitment`?

Reply:
commitment should move, its part of the signing